# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/io/nsLocalFileWin.cpp
# Commit: 25b264ceed3d
# Full Hash: 25b264ceed3d448fb2c8cbee0662baa03303b48e
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2025-10-06 08:04 +0000
# Regressor Bug: 1980886
# File Overlap Count: 1
# Overlapping Files: xpcom/io/nsLocalFileWin.cpp
# Description:
#   Bug 1980886 p1 - Refactor CopySingleFile to better reflect its usage and behaviour. r=yjuglaret
# ==============================================================================

diff -r 4a26473354f3 -r 25b264ceed3d xpcom/io/nsLocalFileWin.cpp
--- a/xpcom/io/nsLocalFileWin.cpp	Mon Oct 06 10:48:11 2025 +0300
+++ b/xpcom/io/nsLocalFileWin.cpp	Mon Oct 06 08:04:24 2025 +0000
@@ -1740,9 +1740,9 @@
   return true;
 }
 
-nsresult nsLocalFile::CopySingleFile(nsIFile* aSourceFile, nsIFile* aDestParent,
-                                     const nsAString& aNewName,
-                                     uint32_t aOptions) {
+nsresult nsLocalFile::MoveOrCopyAsSingleFileOrDir(nsIFile* aDestParent,
+                                                  const nsAString& aNewName,
+                                                  uint32_t aOptions) {
   nsresult rv = NS_OK;
   nsAutoString filePath;
 
@@ -1762,19 +1762,19 @@
 
   if (aNewName.IsEmpty()) {
     nsAutoString aFileName;
-    aSourceFile->GetLeafName(aFileName);
+    GetLeafName(aFileName);
     destPath.Append(aFileName);
   } else {
     destPath.Append(aNewName);
   }
 
   if (aOptions & FollowSymlinks) {
-    rv = aSourceFile->GetTarget(filePath);
+    rv = GetTarget(filePath);
     if (filePath.IsEmpty()) {
-      rv = aSourceFile->GetPath(filePath);
+      rv = GetPath(filePath);
     }
   } else {
-    rv = aSourceFile->GetPath(filePath);
+    rv = GetPath(filePath);
   }
 
   if (NS_FAILED(rv)) {
@@ -1782,11 +1782,8 @@
   }
 
 #ifdef DEBUG
-  nsCOMPtr<nsILocalFileWin> srcWinFile = do_QueryInterface(aSourceFile);
-  MOZ_ASSERT(srcWinFile);
-
   bool srcUseDOSDevicePathSyntax;
-  srcWinFile->GetUseDOSDevicePathSyntax(&srcUseDOSDevicePathSyntax);
+  GetUseDOSDevicePathSyntax(&srcUseDOSDevicePathSyntax);
 
   nsCOMPtr<nsILocalFileWin> destWinFile = do_QueryInterface(aDestParent);
   MOZ_ASSERT(destWinFile);
@@ -1993,7 +1990,7 @@
     if (!aParentDir) {
       aOptions |= SkipNtfsAclReset;
     }
-    rv = CopySingleFile(this, newParentDir, aNewName, aOptions);
+    rv = MoveOrCopyAsSingleFileOrDir(newParentDir, aNewName, aOptions);
     done = NS_SUCCEEDED(rv);
     // If we are moving a directory and that fails, fallback on directory
     // enumeration.  See bug 231300 for details.
@@ -2247,7 +2244,7 @@
     options |= SkipNtfsAclReset;
   }
   // Move single file, or move a directory
-  return CopySingleFile(this, targetParentDir, aNewName, options);
+  return MoveOrCopyAsSingleFileOrDir(targetParentDir, aNewName, options);
 }
 
 NS_IMETHODIMP